### PR TITLE
Clean up flax E2E tests

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Clean up JAX/FLAX E2E model tests as mentioned in the [guidelines](http://shortn/_IcrRByl25f): 

* Model relevance
  * All model rank high in HF tasks, except one in `mnist.libsonnet`. In the [source code](https://github.com/google/flax/tree/main/examples/mnist), it says a `simple convolutional network on the MNIST dataset`. So I deleted it at this moment.
* Test type on TPUs
  * Generally, functional tests run on v2-8, v3-8, and/or v4-8
  * Generally, conv tests run on v2-32, v3-32, and/or v4-32

This change deletes 10 tests and updates 5 tests (previous total: 33, new total: 23). More details on each test can be found in this [doc](http://shortn/_LBGTFLP4SM).

# Tests

Manually run one-shot for updated tests.

**Instruction and/or command lines to reproduce your tests:** 

```
./scripts/run-oneshot.sh -t <test_name>
```

**List links for your tests (use go/shortn-gen for any internal link):** 
* [hf-bert-mrpc-conv-v4-32](http://shortn/_o7m1P42YQz): update from v4-8 to v4-32 for conv test
* [hf-bert-mnli-conv-v4-32](http://shortn/_T357V9oFWX): update from v4-8 to v4-32 for conv test
* [hf-vit-conv-v3-32](http://shortn/_tk9Xoku2Yg): update from func to conv test for v3-32
* [hf-bart-conv-v3-32](http://shortn/_a81TH48Nij): update from func to conv test for v3-32
* [wmt-conv-v3-32](http://shortn/_sjxXUkmxxT): update from v3-8 to v3-32 for conv test. Also update `num_train_steps` from 100_000 to 70_000 as noticed the model has converged and free up resource earlier. This shorts test from ~15 hours to ~9 hours.
 
1) training 100_000 steps
```
[99999] train_accuracy=0.6922277212142944, train_learning_rate=0.00020050253078807145, train_loss=1.4982165098190308
[99999] eval_accuracy=0.7217434048652649, eval_loss=1.2920492887496948
[99999] bleu=28.581661
```
 
2) training 70_000 steps
```
[69999] train_accuracy=0.7108972668647766, train_learning_rate=0.00023990562476683408, train_loss=1.3768692016601562
[69999] eval_accuracy=0.7240751385688782, eval_loss=1.2744629383087158
[69999] bleu=29.303039
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by will-cromar** [comment]
Thanks for the cleanup.

Do you think MNIST has value as a smoke test because it's so simple, even though it's not a real customer use case? It's up to you (or whoever monitors these tests) whether it's worth keeping

File: tests/jax/latest/hf-glue.libsonnet [hunk 1]
```diff
@@ -96,10 +98,13 @@ local utils = import 'templates/utils.libsonnet';
...
+  local v4_32 = {
+    accelerator: tpus.v4_32,
+  },
 
   configs: [
-    hf_bert_common + mnli + convergence + v4 + v4_8 + timeouts.Hours(10),
-    hf_bert_common + mrpc + convergence + v4 + v4_8 + timeouts.Hours(1),
+    hf_bert_common + mnli + convergence + v4 + v4_32,
+    hf_bert_common + mrpc + convergence + v4 + v4_32,
     hf_bert_common + mnli + functional + v4 + v4_8,
```
[the remark this answers is not in the document]
Comment (RissyRan) [reply tests/jax/latest/hf-glue.libsonnet:108]: Thanks Will! Good suggestion, and will update it. 

For the MNIST, I chatted with Susie, and she is fine to remove it as no specific use case/reason when adding it. I slightly prefer to remove it at this moment. But you bring up a quite good point about dependency of tests, i.e. run smoke tests first, and then trigger E2E model tests if smoke tests pass. But unfortunately, we don't have this mechanism yet.

[the remark this answers is not in the document]
Comment (RissyRan) [reply tests/jax/latest/hf-glue.libsonnet]: Done. Tested in [flax-latest-hf-bert-mnli-func-v4-8](http://shortn/_zWMLdDfbA2) with same configs in yaml.

**Review by will-cromar — APPROVED** [review]
Thanks!

**Pull request merged**